### PR TITLE
Expose metrics for `pending acquire` operation latency

### LINKED ISSUE #2946: Metrics Request - Request Pending Wait Time
One of the metrics provided is `reactor.netty.connection.provider.pending.connections` which gauges the number of requests that are waiting for a connection however what we are lacking is a way to know how much time they are waiting on the queue.

## Motivation
The time a request is waiting for a connection is one of the metrics that could indicate clearly a problem with load or networking.

## Desired solution
As part of the existing metrics, we would like a new metric, something like `reactor.netty.connection.provider.pending.connection.time`, of type Timer that will indicate the time a request spent on the queue waiting for a connection.

## Considered alternatives
We have some logging around the HTTP client request however having the metric inside Reactor Netty will be the the closes to the code that really performs the call.

### PR DESCRIPTION
**Pull request by violetagg**
Fixes #2946

### CODE REVIEW

**Comment by violetagg** [comment]
The failed tests on CI are not related to this change.

File: reactor-netty-core/src/main/java/reactor/netty/resources/MicrometerPoolMetricsRecorder.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+				Timer.builder(PENDING_CONNECTIONS_TIME.getName())
+				     .tags(Tags.of(ID.asString(), id, REMOTE_ADDRESS.asString(), formatSocketAddress(remoteAddress),
+				             NAME.asString(), poolName, STATUS.asString(), SUCCESS))
+				     .register(REGISTRY);
+		pendingErrorTimer =
+				Timer.builder(PENDING_CONNECTIONS_TIME.getName())
+				     .tags(Tags.of(ID.asString(), id, REMOTE_ADDRESS.asString(), formatSocketAddress(remoteAddress),
+				             NAME.asString(), poolName, STATUS.asString(), ERROR))
+				     .register(REGISTRY);
+	}
```
Comment (pderop) [inline reactor-netty-core/src/main/java/reactor/netty/resources/MicrometerPoolMetricsRecorder.java:52]:
```suggestion
	MicrometerPoolMetricsRecorder(String id, String poolName, SocketAddress remoteAddress) {
		pendingSuccessTimer = buildTimer(id, poolName, remoteAddress, SUCCESS);
		pendingErrorTimer = buildTimer(id, poolName, remoteAddress, ERROR);
	}

	Timer buildTimer(String id, String poolName, SocketAddress remoteAddress, String status) {
		return Timer.builder(PENDING_CONNECTIONS_TIME.getName())
				.tags(Tags.of(ID.asString(), id, REMOTE_ADDRESS.asString(), formatSocketAddress(remoteAddress),
						NAME.asString(), poolName, STATUS.asString(), status))
				.register(REGISTRY);
	}
```

What do you think about initializing both timers with a common `buildTimer` method ?

File: reactor-netty-http/src/main/java/reactor/netty/http/client/MicrometerPoolMetricsRecorder.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+				Timer.builder(PENDING_STREAMS_TIME.getName())
+				     .tags(Tags.of(ID.asString(), id, REMOTE_ADDRESS.asString(), formatSocketAddress(remoteAddress),
+				             NAME.asString(), poolName, STATUS.asString(), SUCCESS))
+				     .register(REGISTRY);
+		pendingErrorTimer =
+				Timer.builder(PENDING_STREAMS_TIME.getName())
+				     .tags(Tags.of(ID.asString(), id, REMOTE_ADDRESS.asString(), formatSocketAddress(remoteAddress),
+				             NAME.asString(), poolName, STATUS.asString(), ERROR))
+				     .register(REGISTRY);
+	}
```
Comment (pderop) [inline reactor-netty-http/src/main/java/reactor/netty/http/client/MicrometerPoolMetricsRecorder.java:52]:
```suggestion
	MicrometerPoolMetricsRecorder(String id, String poolName, SocketAddress remoteAddress) {
		pendingSuccessTimer = buildTimer(id, poolName, remoteAddress, SUCCESS);
		pendingErrorTimer = buildTimer(id, poolName, remoteAddress, ERROR);
	}

	Timer buildTimer(String id, String poolName, SocketAddress remoteAddress, String status) {
		return Timer.builder(PENDING_STREAMS_TIME.getName())
				.tags(Tags.of(ID.asString(), id, REMOTE_ADDRESS.asString(), formatSocketAddress(remoteAddress),
						NAME.asString(), poolName, STATUS.asString(), status))
				.register(REGISTRY);
	}
```

What do you think about initializing both timers with a common buildTimer method ?

File: reactor-netty-core/src/main/java/reactor/netty/resources/MicrometerPoolMetricsRecorder.java (hunk 1 above)
Comment (violetagg) [reply reactor-netty-core/src/main/java/reactor/netty/resources/MicrometerPoolMetricsRecorder.java]: fixed

File: reactor-netty-http/src/main/java/reactor/netty/http/client/MicrometerPoolMetricsRecorder.java (hunk 2 above)
Comment (violetagg) [reply reactor-netty-http/src/main/java/reactor/netty/http/client/MicrometerPoolMetricsRecorder.java]: fixed

**Review by pderop — APPROVED** [review]

**Comment by violetagg** [comment]
@pderop Thanks for the review!

**Pull request merged**